### PR TITLE
Add pontos-update-header option to ignore files

### PR DESCRIPTION
**Pull request by timopollmeier**
**What**:
The script can be passed a file path or check the directory given with
the --directory option for a .pontos-header-ignore file.
All files matching the paths or globs listed in this filed will be
skipped when adding or updating the headers.

**Why**:
To allow skipping build directories and other files that should not have a header.

**How**:
By running the script on a directory with a .pontos-header-ignore file
that has both a normal path and a recursive glob.

**Checklist**:

- [x] Tests
- [x] [CHANGELOG](https://github.com/greenbone/pontos/blob/master/CHANGELOG.md) Entry
- [ ] Documentation

### CODE REVIEW

File: pontos/updateheader/updateheader.py [hunk 1]
```diff
@@ -278,7 +321,10 @@ def main() -> None:
 
     for file in files:
         try:
-            _update_file(file=file, regex=regex, args=args)
+            if os.path.abspath(str(file)) in exclude_list:
+                print(f"{file}: Matches exclusion list.")
```
Comment (y0urself) [inline pontos/updateheader/updateheader.py:325]:
```suggestion
                print(f"{file}: Ignoring file from exclusion list.")
```
```suggestion
                print(f"{file}: Matches exclusion list.")
```

**Review by y0urself — APPROVED** [review]

File: pontos/updateheader/updateheader.py [hunk 2]
```diff
@@ -195,6 +195,34 @@ def _update_file(
...
+        glob(os.path.join(directory, line.replace('\n', '')), recursive=True)
+        for line in exclude_lines
+        if line
+    ]
+
+    return [
+        os.path.abspath(path)
+        for glob_paths in expanded_globs
+        for path in glob_paths
+    ]
```
Comment (bjoernricks) [inline pontos/updateheader/updateheader.py:223]: Please don't use os.path anymore. All code must use pathlib now. pathlib is much easier to handle.

File: pontos/updateheader/updateheader.py [hunk 3]
```diff
@@ -195,6 +195,34 @@ def _update_file(
...
 
+def _get_exclude_list(exclude_file, directory):
+    """Tries to get the list of excluded files.
+    If a file is given, it will be used. Otherwise the given directory will be
+    checked for the default file.
+    """
+    if exclude_file is None:
+        path = os.path.join(directory, ".pontos-header-ignore")
+        try:
+            exclude_file = open(path, "r")
```
Comment (bjoernricks) [inline pontos/updateheader/updateheader.py:206]: open should never be used without a with statement. Here either [Path.open](https://docs.python.org/3/library/pathlib.html#pathlib.Path.open) or [Path.read_text](https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_text) should be used.

**Review by bjoernricks — CHANGES_REQUESTED** [review]

File: pontos/updateheader/updateheader.py [hunk 4]
```diff
@@ -195,6 +195,34 @@ def _update_file(
         raise e
 
 
+def _get_exclude_list(exclude_file, directory):
+    """Tries to get the list of excluded files.
+    If a file is given, it will be used. Otherwise the given directory will be
+    checked for the default file.
+    """
+    if exclude_file is None:
+        path = os.path.join(directory, ".pontos-header-ignore")
```
Comment (y0urself) [inline pontos/updateheader/updateheader.py:204]:
```suggestion
        path = Path(directory) / ".pontos-header-ignore"
```

**Review by bjoernricks — APPROVED** [review]

**Pull request merged**